### ISSUE TITLE
Bug(prefer-string-schema-with-trim): fires on `z.record()` key schema, encouraging silent data loss

### ISSUE THREAD

**Issue by better-salmon** [marcalexiei/eslint-plugin-zod#242]
### 🧩 Description

`prefer-string-schema-with-trim` warns on every `z.string()` without `.trim()`, including when it's used as the **key schema** of `z.record()`. Following this warning introduces a bug: `.trim()` on record keys is a transform that silently mutates keys and can cause key collisions with data loss.

### 🪜 How to reproduce

1. Enable `prefer-string-schema-with-trim` (included in `recommended` config)
2. Write this correct code:

```ts
const schema = z.record(z.string(), z.unknown());
```

3. The rule warns and suggests changing it to:

```ts
const schema = z.record(z.string().trim(), z.unknown());
```

4. After "fixing" the warning, parsing this input:

```ts
schema.parse({
  " userId ": "abc-123",
  "userId": "def-456",
});
// → { userId: "def-456" }
// "abc-123" is silently lost — 2 keys collapsed into 1
```

### ✅ Expected behavior

`prefer-string-schema-with-trim` should **not** report when `z.string()` appears as the first argument (key schema) of `z.record()`. Transforms on record keys change the shape of the output object in ways that pure validators do not, and the user almost certainly intended pass-through key behavior.

### 💻 System info

```shell
Binaries:
  Node: 24.14.1 - /Users/name/.vite-plus/js_runtime/node/24.14.1/bin/node
  npm: 11.11.0 - /Users/name/.vite-plus/js_runtime/node/24.14.1/bin/npm
  pnpm: 10.33.0 - /Users/name/.vite-plus/js_runtime/node/24.14.1/bin/pnpm
  bun: 1.3.12 - /Users/name/.bun/bin/bun
  Deno: 2.7.12 - /Users/name/.deno/bin/deno
npmPackages:
  eslint: catalog: => 10.2.0
  eslint-plugin-zod: catalog: => 3.5.3
```

**Comment by marcalexiei** [comment]
Fix available in [v3.5.4](https://github.com/marcalexiei/eslint-plugin-zod/releases/tag/v3.5.4).

Thanks for the report!
